commcare_connect/opportunity/tw_views.py

```diff
@@ -3717,4 +3724,45 @@ def opportunity_worker_payment(request, org_slug=None, opp_id=None):
         },
     ]
     table = OpportunityWorkerPaymentTable(data)
-    return render(request, "tailwind/pages/opportunity_worker_extended.html", {"header_title": "Worker", "kpi":user_kpi, "tab_name": "Payment", "table": table })    
+    return render(request, "tailwind/pages/opportunity_worker_extended.html", {"header_title": "Worker", "kpi":user_kpi, "tab_name": "Payment", "table": table })
+
+
+
+class OpportunityListView(OrganizationUserMixin, SingleTableMixin, TemplateView):
+    template_name = "tailwind/pages/opportunities_list.html"
+    table_class = OpportunitiesListViewTable
+    paginate_by = 15
+
+    allowed_sort_columns = [
+        'program',
+        'start_date',
+        'end_date',
+        'pending_invites',
+        'inactive_workers',
+        'pending_approvals',
+        'payments_due',
+        'status'
+    ]
+
+    def get_table_data(self):
+        org = self.request.org
+        return get_opportunity_list_data(org)
+
+    def get_validated_order_by(self):
+        requested_order = self.request.GET.get('sort', 'start_date')
+
+        is_descending = requested_order.startswith('-')
+        column = requested_order[1:] if is_descending else requested_order
+        return requested_order if column in self.allowed_sort_columns else 'start_date'
+
+    def get_table(self, **kwargs):
+        table = super().get_table(**kwargs)
+
+        validated_order = self.get_validated_order_by()
+
+        if validated_order:
+            table.order_by = validated_order
+        else:
+            table.order_by = ("-status", "start_date", "end_date")
+
+        return table
```

Comment:
@hemant10yadav Why do we need to manage this sorting? Django-tables already does.